### PR TITLE
Update release-3.4 and release-3.5 CHANGELOG with go version bump

### PR DESCRIPTION
**Pull request by henrybear327**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Reference: https://github.com/etcd-io/etcd/issues/18419

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
LGTM

thx

File: CHANGELOG/CHANGELOG-3.5.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 
 ## v3.5.16 (TBC)
 
-<hr>
+- Compile binaries using [go 1.21.13](https://github.com/etcd-io/etcd/pull/18421).
```
Comment (ivanvc) [inline CHANGELOG/CHANGELOG-3.5.md:9]: @henrybear327, I think we always keep the horizontal rule at the end of the release notes for that particular release. I think you're also missing adding the Dependencies header :smiley: 

```suggestion
### Dependencies
- Compile binaries using [go 1.21.13](https://github.com/etcd-io/etcd/pull/18421).

<hr>
```

**Review by ivanvc — COMMENTED** [review]
Thanks for the PR, @henrybear327. I left a comment in the 3.5 CHANGELOG :)

File: CHANGELOG/CHANGELOG-3.5.md (hunk 1 above)
Comment (henrybear327) [reply CHANGELOG/CHANGELOG-3.5.md:10]: Thanks for catching this.

I reused the style from https://github.com/etcd-io/bbolt/blob/main/CHANGELOG/CHANGELOG-1.3.md#other when I did the update. Sorry for this...

**Review by ivanvc — APPROVED** [review]
LGTM on green. Thanks, @henrybear327.

**Pull request merged**